design/status-design-update.md

```diff
@@ -0,0 +1,1166 @@
+      includes:
+      - name: ios
+        namespace: default
+        conditions:
+        - prefix: /
+        - header:
+            name: x-os
+            contains: ios
+        status:
+          currentStatus: Valid
+          description: Valid include
+        routes:
+        - services:
+          - name: ios
+            port: 80
+          conditions:
+          - prefix: /
+          - header:
+              name: x-os
+              contains: ios
+          status:
+            currentStatus: Valid
+            description: Valid route
+  appliedConditions:
+  - fqdn: www.widgetsandsprockets.com
+    includes:
+    - name: defaultsite
+      namespace: default
+      conditions:
+      - prefix: /
+      includes:
+      - name: ios
+        namespace: default
+        conditions:
+        - prefix: /
+        - header:
+            name: x-os
+            contains: ios
+---
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: defaultbackend
+  namespace: default
+spec:
+  routes:
+  - services:
+    - name: defaultbackend
+      port: 80
+status:
+  currentStatus: valid
+  description: valid HTTPProxy
+  includes: {}
+  routes:
+  - fqdn: www.widgetsandsprockets.com
+    conditions:
+    - prefix: /
+    services:
+    - name: defaultbackend
+      port: 80
+    includePath: "root:www.widgetsandsprockets.com,defaultsite,default"
+  proxyGraph:
+  - fqdn: www.widgetsandsprockets.com
+    includes:
+    - name: defaultsite
+      namespace: default
+      conditions:
+      - prefix: /
+      status:
+        currentStatus: Valid
+        description: Valid include
+      includes:
+      - name: default
+        namespace: default
+        conditions:
+        - prefix: /
+        status:
+          currentStatus: Valid
+          description: Valid include
+       routes:
+        - services:
+          - name: defaultbackend
+            port: 80
+          conditions:
+          - prefix: /
+          status:
+            currentStatus: Valid
+            description: Valid route
+  appliedConditions:
+  - fqdn: www.widgetsandsprockets.com
+    includes:
+    - name: defaultsite
+      namespace: default
+      conditions:
+      - prefix: /
+      includes:
+      - name: default
+        namespace: default
+        conditions:
+        - prefix: /
+```
+
+### Invalid: Root with no FQDN
+
+In this case, the root HTTPProxy does not include a FQDN key, so the included HTTPProxy will be left orphaned.
+
+I think that whether or not `proxyGraph` should be filled out in the case that the HTTPProxy is invalid at the root level is probably very dependent on exactly how this is implemented.
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: invalidParent
+  namespace: roots
+spec:
+  virtualhost:
+  includes:
+  - name: validChild
+    namespace: roots
+    conditions:
+      - prefix: /foo
+status:
+  currentStatus: invalid
+  description: Spec.VirtualHost.Fqdn must be specified
```

Comment:
This can't happen, the only way to know what a "root" object is by the presence of the fqdn. If it doesn't have it, then it's child and would be orphaned.

Reply:
Okay, thanks. I'll updated.